prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -275,39 +313,6 @@ def get_function_bullet_point_list(agent: Agent, model: str) -> str:
         with col3:
             st.caption(" \- OR -")
         with col4:
-            user_reasoning = st.chat_input("Add reasoning...")
-
-# Execution
```

Comment:
I'm confused!

Here: https://github.com/gnosis/prediction-market-agent/pull/222#discussion_r1635226431 you say

> Regarding separating Execution and Layout, also fine and fits nicely with my changes from above.

but now execution is all inline with layout!!

Reply:
Please see latest state of app.py -> https://github.com/gnosis/prediction-market-agent/blob/1a3e9421191e461c696b05c1117c06e4bd954bb1/prediction_market_agent/agents/microchain_agent/app.py#L188

What I meant was: from line 188, we only have layout elements that reference st.state_layout.<VARIABLE>. All the functions are above and do not get called.
Are you ok with this current structure?

Reply:
What's `st.state_layout`?

For the balance lag issue, here's what I had in mind for a fix https://github.com/gnosis/prediction-market-agent/pull/247. Can we do something similar here for the system prompt?

Reply:
Here's the equivalent kind of change to fix the system prompt/bootstrap lag https://github.com/gnosis/prediction-market-agent/pull/254

Reply:
I merged `app.py` from main, which comes from the PR you mentioned. I believe this solves this issue.